internal/cri/opts/spec_opts.go

```diff
@@ -118,6 +118,29 @@ func WithAnnotation(k, v string) oci.SpecOpts {
 	}
 }
 
+// WithAffinityCPUs sets the CPU affinity values in runtime spec for windows.
+func WithAffinityCPUs(affinityCPUs []*runtime.WindowsCpuGroupAffinity) oci.SpecOpts {
+	return func(_ context.Context, _ oci.Client, c *containers.Container, s *runtimespec.Spec) error {
+		// We would expect the Windows and Resources struct to be setup by now
+		if affinityCPUs == nil || s.Windows == nil || s.Windows.Resources == nil {
+			return nil
+		}
+		if s.Windows.Resources.CPU == nil {
+			s.Windows.Resources.CPU = &runtimespec.WindowsCPUResources{}
+		}
+		if s.Windows.Resources.CPU.Affinity == nil {
+			s.Windows.Resources.CPU.Affinity = make([]runtimespec.WindowsCPUGroupAffinity, len(affinityCPUs))
+		}
+		for i, affinity := range affinityCPUs {
+			s.Windows.Resources.CPU.Affinity[i] = runtimespec.WindowsCPUGroupAffinity{
```

Comment:
This breaks if `s.Windows.Resources.CPU.Affinity` was not nil.
Perhaps use append and just set the slice cap a L132.